File: FantasyRadio/src/main/java/ru/sigil/fantasyradio/utils/RadioStreamFactory.java
package ru.sigil.fantasyradio.utils;

import android.content.Context;

import java.util.HashMap;
import java.util.Map;

import javax.inject.Inject;

import ru.sigil.bassplayerlib.StreamFormat;
import ru.sigil.fantasyradio.R;

/**
 * Created by
 * namelessone on 11.03.17.
 */

public class RadioStreamFactory {
    private final Map<Bitrate, String> bitratesMap;

    @Inject
    public RadioStreamFactory(Context context) {
        bitratesMap = new HashMap<>();
        bitratesMap.put(Bitrate.aac_16, context.getString(R.string.stream_url_AAC16));
        bitratesMap.put(Bitrate.mp3_32, context.getString(R.string.stream_url_MP332));
        bitratesMap.put(Bitrate.mp3_96, context.getString(R.string.stream_url_MP396));
        bitratesMap.put(Bitrate.aac_112, context.getString(R.string.stream_url_AAC112));
    }

    public RadioStream createDefaultStream() {
        return createStreamWithBitrate(Bitrate.aac_16);
    }

    public RadioStream createStreamWithBitrate(Bitrate bitrate) {
        StreamFormat streamFormat;
        switch (bitrate)
        {
            case aac_16:
            case aac_112:
                streamFormat = StreamFormat.aac;
                break;
            case mp3_96:
            case mp3_32:
            default:
                streamFormat = StreamFormat.mp3;
                break;
        }
        return new RadioStream(bitrate, bitratesMap.get(bitrate), streamFormat);
    }
}

File: FantasyRadio/src/main/kotlin/ru/sigil/fantasyradio/utils/RadioStreamFactory.kt
package ru.sigil.fantasyradio.utils

import android.content.Context
import ru.sigil.fantasyradio.R
import javax.inject.Inject
import ru.sigil.bassplayerlib.StreamFormat

/**
 * Created by namelessone
 * on 08.12.18.
 */
class RadioStreamFactory @Inject constructor(context: Context) : IRadioStreamFactory {
    private val bitRatesMap: Map<Bitrate, String>

    init {
        bitRatesMap = HashMap()
        bitRatesMap.put(Bitrate.AAC_16, context.getString(R.string.stream_url_AAC16))
        bitRatesMap.put(Bitrate.MP3_32, context.getString(R.string.stream_url_MP332))
        bitRatesMap.put(Bitrate.MP3_96, context.getString(R.string.stream_url_MP396))
        bitRatesMap.put(Bitrate.AAC_112, context.getString(R.string.stream_url_AAC112))
    }

    override fun createDefaultStream(): RadioStream {
        return createStreamWithBitrate(Bitrate.AAC_16)
    }

    override fun createStreamWithBitrate(bitrate: Bitrate): RadioStream {
        val streamFormat: StreamFormat = when (bitrate) {
            Bitrate.AAC_16, Bitrate.AAC_112 -> StreamFormat.AAC
            Bitrate.MP3_96, Bitrate.MP3_32 -> StreamFormat.MP3
        }
        return RadioStream(bitrate, bitRatesMap[bitrate]!!, streamFormat)
    }
}